internal/dataaccess/expense_dao.go

```diff
@@ -0,0 +1,161 @@
+	var exists bool
+	err := e.conn.QueryRow(context.Background(),
+		`SELECT EXISTS(SELECT 1 FROM public.Expenses WHERE id = $1)`, id).Scan(&exists)
+	return exists, err
+}
+
+// Adds a promoted expense (need to validate).
+func (e *ExpenseDAO) CreateExpense(exp entities.Expense) error {
+	query := `INSERT INTO public.Expenses (Date, Description, Amount) VALUES (@Date, @Description, @Amount)`
+
+	args := pgx.NamedArgs{
+		"Date":        exp.Date,
+		"Description": exp.Desc,
+		"Amount":      exp.Amount,
+	}
+
+	_, err := e.conn.Exec(context.Background(), query, args)
+	if err != nil {
+		return fmt.Errorf("error inserting expense: %w", err)
+	}
+	return nil
+}
+
+// Returns an array of all expenses available.
+func (e *ExpenseDAO) GetAllExpensesFromDb() ([]entities.Expense, error) {
+	query := `SELECT * FROM public.Expenses`
+
+	rows, err := e.conn.Query(context.Background(), query)
+	if err != nil {
+		fmt.Println("Error querying the table")
+		return nil, err
+	}
+
+	defer rows.Close()
+
+	var expenses []entities.Expense
+
+	for rows.Next() {
+		var exp entities.Expense
+		err := rows.Scan(&exp.ID, &exp.Date, &exp.Desc, &exp.Amount)
+		if err != nil {
+			fmt.Println("Error fetching expense details")
+			return expenses, err
+		}
+		expenses = append(expenses, exp)
+	}
+	return expenses, nil
+}
+
+// Replaces expense with promoted id with new given expense.
+func (e *ExpenseDAO) UpdateExpense(id int, exp entities.Expense) error {
+	query := `
+		UPDATE public.Expenses
+		SET Date = @Date, Description = @Description, Amount = @Amount 
+		WHERE id = @id
+	`
+
+	args := pgx.NamedArgs{
+		"id":          id,
+		"Date":        exp.Date,
+		"Description": exp.Desc,
+		"Amount":      exp.Amount,
+	}
+
+	_, err := e.conn.Exec(context.Background(), query, args)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+// Deletes an expense with promoted id.
+func (e *ExpenseDAO) DeleteExpense(id int) error {
+	query := `
+	DELETE FROM public.Expenses WHERE id = @id`
+
+	args := pgx.NamedArgs{
+		"id": id,
+	}
+
+	_, err := e.conn.Exec(context.Background(), query, args)
+	if err != nil {
+		fmt.Println(err)
+		return err
+	}
+	return nil
+}
+
+// to help count sum of array.
+func sumOfArray(i []int) int {
+	summ := 0
+	for _, v := range i {
+		summ += v
+	}
+	return summ
+}
+
+// Returns an amount of all expenses.
+func (e *ExpenseDAO) Summary() (int, error) {
+	query := `
+	SELECT e.Amount FROM public.Expenses as e
+	`
+
+	rows, err := e.conn.Query(context.Background(), query)
+
+	if err != nil {
+		fmt.Println("Error getting summary")
+		return -1, err
+	}
+	defer rows.Close()
+
+	var amounts []int
+
+	for rows.Next() {
+		var amount int
+		err := rows.Scan(&amount)
+		if err != nil {
+			fmt.Println("Error getting one amount")
+			return sumOfArray(amounts), err
+		}
+		amounts = append(amounts, amount)
+	}
+	return sumOfArray(amounts), nil
+}
+
+// Wipes out db (and resets id's too!).
+func (e *ExpenseDAO) ResetDatabase() error {
```

Comment:
```suggestion
func (e *ExpenseDAO) ResetExpenses() error {
```
Тут я тоже максимизирую педантизм. Вот прямо сейчас наша база данных = одна табличка затрат. Но по ходу дальнейшей разработки у нас могут появиться и другие таблички в базе.

Так вот, наш DAO отвечает только за работу с тратами, а значит и ресетнуть может только их. Поэтому предлагаю назвать метод так, чтобы было понятно, что именно он ресетает.

А если мы прямо хотим иметь метод для ресета всей базы целиком, то ему не место в `ExpenseDAO`. Под него стоило бы выдумать какой-то отдельный объект (в этом же пакете `dataaccess`. Навскидку - такой объект можно обозвать как `DataManager` или `ResetManager`, и положить метод для ресета всех табличек в этот объект.